apps/passport-client/src/dispatch.ts

```diff
@@ -942,3 +949,16 @@ async function promptToAgreePrivacyNotice(state: AppState, update: ZuUpdate) {
     });
   }
 }
+
+/**
+ * Merge in a PCD collection from an import of backed-up data.
+ */
+function mergeImport(
+  state: AppState,
+  update: ZuUpdate,
+  collection: PCDCollection,
+  filter: (pcd: PCD) => boolean
```

Comment:
I think the construction of the `filter` closure needs to be moved here in order to be safe.  It captures the PCDCollection for uuid comparisons, and until you get here there's no guarantee it hasn't been replaced with another one via update().

Alternately maybe the functionality is simple enough it should just be encoded in the options arg below?  Filtering duplicates certainly can be.  Special-casing the semaphore ID seems like it can be as well, though I don't know the current dependency situation.

That being said, the filter closure is something I might need anyway to adapt this merge for sync use, so go whichever way you think best for this diff and I can follow up.  I need more time to think through exactly how deep I want to go on sync-merge.  E.g. I feel like I want to explicitly detect when I encounter two PCDs with the same ID but different contents and report it.  However I'm not sure if there's anything to do in that situation other than keep one of them, so maybe it's not worth detecting at all until a later iteration.

Reply:
Encoding it in the options arg forces `PCDCollection` to be opinionated about semaphore identities, which it currently is not. Injecting the special-case logic for handling semaphore identities keeps `PCDCollection` "cleaner" This is similar to a trade-off I encountered earlier when dealing with feed credentials, which in practice are always semaphore signatures but in principle should not be tied to that particular implementation.